airflow/utils/timezone.py

```diff
@@ -175,6 +175,16 @@ def parse(string: str, timezone=None) -> DateTime:
     return pendulum.parse(string, tz=timezone or TIMEZONE, strict=False)  # type: ignore
 
 
+@overload
+def coerce_datetime(v: None) -> None:
+    ...
+
+
+@overload
+def coerce_datetime(v: Union[dt.datetime, DateTime]) -> DateTime:
+    ...
```

Comment:
`pendulum.DateTime` is a stdlib `datetime` subclass, so this is equivalent to

```suggestion
def coerce_datetime(v: dt.datetime) -> DateTime:
    ...
```

same goes for the original annotation; it can just be `Optional[dt.datetime]`.

Reply:
Ah good to know - I wasn't aware of that.

Reply:
fixed in f77d88edf